### PR TITLE
updating links from fwlink to real link if exists

### PR DESCRIPTION
**Pull request by eddynaka**
updating links from fwlink to real link if exists - contributes to #3391

### CODE REVIEW

File: docs/framework/net-native/example-handling-exceptions-when-binding-data.md
Comment (mairaw) [inline docs/framework/net-native/example-handling-exceptions-when-binding-data.md]: I'd undo this change. Not sure where to go to download the preview now.

```suggestion
> This topic refers to the .NET Native Developer Preview, which is pre-release software. You can download the preview from the [Microsoft Connect website](https://go.microsoft.com/fwlink/?LinkId=394611) (requires registration).  
```

File: docs/framework/net-native/example-troubleshooting-dynamic-programming.md
Comment (mairaw) [inline docs/framework/net-native/example-troubleshooting-dynamic-programming.md]:
```suggestion
> This topic refers to the .NET Native Developer Preview, which is pre-release software. You can download the preview from the [Microsoft Connect website](https://go.microsoft.com/fwlink/?LinkId=394611) (requires registration).  
```

File: docs/framework/net-native/runtime-directive-policy-settings.md
Comment (mairaw) [inline docs/framework/net-native/runtime-directive-policy-settings.md]:
```suggestion
> This topic refers to the .NET Native Developer Preview, which is pre-release software. You can download the preview from the [Microsoft Connect website](https://go.microsoft.com/fwlink/?LinkId=394611) (requires registration).
```

File: docs/framework/network-programming/networkinformation.md
Comment (mairaw) [inline docs/framework/network-programming/networkinformation.md]: @davidsh @karelz have these NCL samples been ported somewhere?

File: docs/framework/network-programming/pnrp-in-application-development.md
Comment (mairaw) [inline docs/framework/network-programming/pnrp-in-application-development.md]:
```suggestion
 Message passing between PeerToPeer applications is still handled by underlying architectures such as PeerChannel and WCF [Large Data and Streaming](../wcf/feature-details/large-data-and-streaming.md).  
```

File: docs/framework/performance/caching-in-net-framework-applications.md
Comment (mairaw) [inline docs/framework/performance/caching-in-net-framework-applications.md]:
```suggestion
 The .NET Framework also enables you to implement conditional HTTP GET caching. In REST scenarios, a conditional HTTP GET request is often used by services to implement intelligent HTTP caching as described in the [HTTP Specification](https://www.w3.org/Protocols/rfc2616/rfc2616.html). For more information, see [Caching Support for WCF Web HTTP Services](../wcf/feature-details/caching-support-for-wcf-web-http-services.md).  
```

File: docs/framework/performance/etw-events-in-the-common-language-runtime.md
Comment (mairaw) [inline docs/framework/performance/etw-events-in-the-common-language-runtime.md]:
```suggestion
 More information about ETW is available in the [Improve Debugging and Performance Tuning with ETW](https://docs.microsoft.com/archive/msdn-magazine/2007/april/event-tracing-improve-debugging-and-performance-tuning-with-etw) article. Information about Xperf can be found in the entry [Windows Performance Toolkit - Xperf](https://blogs.msdn.microsoft.com/ntdebugging/2008/04/03/windows-performance-toolkit-xperf/) in the NTDebugging blog.  
```

File: docs/framework/performance/exception-thrown-v1-etw-event.md
Comment (mairaw) [inline docs/framework/performance/exception-thrown-v1-etw-event.md]:
```suggestion
|ExceptionFlags|win:UInt16|0x01: HasInnerException (see [CLR ETW Events](clr-etw-events.md) in the Visual Basic documentation).<br /><br /> 0x02: IsNestedException.<br /><br /> 0x04: IsRethrownException.<br /><br /> 0x08: IsCorruptedStateException (indicates that the process state is corrupt; see [Handling Corrupted State Exceptions](https://docs.microsoft.com/archive/msdn-magazine/2009/february/clr-inside-out-handling-corrupted-state-exceptions)).<br /><br /> 0x10: IsCLSCompliant (an exception that derives from <xref:System.Exception> is CLS-compliant; otherwise, it is not CLS-compliant).|  
```

File: docs/framework/reflection-and-codedom/dynamic-language-runtime-overview.md
Comment (mairaw) [inline docs/framework/reflection-and-codedom/dynamic-language-runtime-overview.md]:
```suggestion
- IronRuby. Available as open-source software from the [IronRuby](http://ironruby.net/) website.
```

File: docs/framework/reflection-and-codedom/emitting-dynamic-methods-and-assemblies.md
Comment (mairaw) [inline docs/framework/reflection-and-codedom/emitting-dynamic-methods-and-assemblies.md]:
```suggestion
Another useful resource for working with metadata and MSIL is the Common Language Infrastructure (CLI) documentation, especially "Partition II: Metadata Definition and Semantics" and "Partition III: CIL Instruction Set". The documentation is available online at the [Ecma Web site](https://www.ecma-international.org/publications/standards/Ecma-335.htm).  
```

File: docs/framework/reflection-and-codedom/reflection-for-windows-store-apps.md
Comment (mairaw) [inline docs/framework/reflection-and-codedom/reflection-for-windows-store-apps.md]:
```suggestion

Starting with the .NET Framework 4.5, the .NET Framework includes a set of reflection types and members for use in [!INCLUDE[win8_appname_long](../../../includes/win8-appname-long-md.md)] apps. These types and members are available in the full .NET Framework as well as in the .NET for Windows Store apps. This document explains the major differences between these and their counterparts in the .NET Framework 4 and earlier versions.  
```

File: docs/framework/reflection-and-codedom/reflection-for-windows-store-apps.md
Comment (mairaw) [inline docs/framework/reflection-and-codedom/reflection-for-windows-store-apps.md]: let's remove this link

```suggestion
```

File: docs/framework/resources/retrieving-resources-in-desktop-apps.md
Comment (mairaw) [inline docs/framework/resources/retrieving-resources-in-desktop-apps.md]:
```suggestion

When you work with localized resources in .NET Framework desktop apps, you should ideally package the resources for the default or neutral culture with the main assembly and create a separate satellite assembly for each language or culture that your app supports. You can then use the <xref:System.Resources.ResourceManager> class as described in the next section to access named resources. If you choose not to embed your resources in the main assembly and satellite assemblies, you can also access binary .resources files directly, as discussed in the section [Retrieving Resources from .resources files](#from_file) later in this article.  To retrieve resources in [!INCLUDE[win8_appname_long](../../../includes/win8-appname-long-md.md)] apps, see [Creating and retrieving resources in Windows Store apps](https://docs.microsoft.com/previous-versions/windows/apps/hh694557(v=vs.140)).  
```

File: docs/framework/security/building-my-first-claims-aware-aspnet-web-app.md
Comment (mairaw) [inline docs/framework/security/building-my-first-claims-aware-aspnet-web-app.md]: I would roll back changes under the security folder since I'm actually trying to archive this content in PR #14590

**Review by mairaw — CHANGES_REQUESTED** [review]
Thanks @eddynaka. Left a few comments for you to consider before we can merge this.

File: docs/framework/network-programming/networkinformation.md
Comment (Youssef1313) [reply docs/framework/network-programming/networkinformation.md]: @mairaw, I think this is the NetStat Tool Technology Sample.

https://github.com/KenAdeniji/WordEngineering/blob/23aa5c2629e6db2618afa6f9fa843a4370da18b9/InformationInTransit/ProcessLogic/NetStatTool.cs

Comment (Youssef1313) [reply docs/framework/network-programming/networkinformation.md]: The comment in it exactly matches the Remarks appears in the WayBack machine:

image

Comment (Youssef1313) [reply docs/framework/network-programming/networkinformation.md]: I've also found [this](https://github.com/Mooliecool/windows/tree/master/sdk/samples/FrameworkSamples/NCL), which contains all the samples.

Comment (davidsh) [reply docs/framework/network-programming/networkinformation.md]: >@davidsh @karelz have these NCL samples been ported somewhere?

I am not aware of any porting efforts for those samples. If they are ported, they should be placed in some GitHub repo.

@mairaw do we have an existing GitHub repo for samples?

Comment (mairaw) [reply docs/framework/network-programming/networkinformation.md]: Yes, we host all our samples at https://github.com/dotnet/samples.

Comment (eddynaka) [reply docs/framework/network-programming/networkinformation.md]: Hi all, what should we do in this case? Thanks!

Comment (Youssef1313) [reply docs/framework/network-programming/networkinformation.md]: @eddynaka, I think @mairaw can suggest where the samples need to be ported in the samples repo and just update the link and any occurrences of it to the samples repo. Not sure if it's good idea to add it to samples browser or not.

Comment (mairaw) [reply docs/framework/network-programming/networkinformation.md]: @davidsh do we have these samples hosted internally somewhere? I'd prefer to grab them from official sources.

File: docs/framework/network-programming/networkinformation.md
Comment (mairaw) [inline docs/framework/network-programming/networkinformation.md]: @eddynaka for now let's comment it out the three samples links like this:

```markdown
<!-- to-do: review sample links
- [Network Information Technology Sample](https://go.microsoft.com/fwlink/?LinkID=179564)
- [NetStat Tool Technology Sample](https://go.microsoft.com/fwlink/?LinkID=179562)
- [Ping Client Technology Sample](https://go.microsoft.com/fwlink/?LinkID=179565)
--> 
```

File: docs/framework/network-programming/networkinformation.md
Comment (mairaw) [reply docs/framework/network-programming/networkinformation.md]: I've opened #15913 for us to discuss that so we can unblock this PR

**Comment by eddynaka** [comment]
@mairaw , just pushed another commit commenting that samples.

File: docs/framework/network-programming/peer-name-resolution-protocol.md
Comment (mairaw) [inline docs/framework/network-programming/peer-name-resolution-protocol.md]: comment this one as well?

```suggestion

<!-- to-do: review sample links
- [PeerToPeer Technology Sample](https://go.microsoft.com/fwlink/?LinkID=179571)
-->
```

**Review by mairaw — APPROVED** [review]
Thanks @eddynaka! Let's see if the linting thing is just a temp issue. Otherwise, I'll try a different solution.

**Pull request merged**
